packages/client/tests/functional/issues/13766/tests.ts

```diff
@@ -0,0 +1,88 @@
+import { faker } from '@faker-js/faker'
+// @ts-ignore
+import type { PrismaClient } from '@prisma/client'
+
+import testMatrix from './_matrix'
+
+declare let prisma: PrismaClient
+
+// https://github.com/prisma/prisma/issues/13766
+testMatrix.setupTestSuite(
+  ({ provider }) => {
+    test('should not prevent any updates on a model when updating a field', async () => {
+      const orderId = faker.random.numeric().toString()
+
+      await prisma.order.create({
+        data: {
+          orderId,
+          paid: false,
+          statusMilestones: {
+            create: {
+              status: 'NEW',
+            },
+          },
+        },
+      })
+
+      const updatedOrder = await prisma.order.update({
+        where: {
+          orderId,
+        },
+        data: {
+          paid: true,
+        },
+      })
+
+      expect(updatedOrder).toMatchObject({ orderId, paid: true })
+    })
+
+    test('should prevent updates on a model if any other relation references an field', async () => {
+      const orderId1 = faker.random.numeric().toString()
+      const orderId2 = faker.random.numeric().toString()
+
+      await prisma.order.create({
+        data: {
+          orderId: orderId1,
+          paid: false,
+          statusMilestones: {
+            create: {
+              status: 'NEW',
+            },
+          },
+        },
+      })
+
+      await prisma.order.create({
+        data: {
+          orderId: orderId2,
+          paid: false,
+          statusMilestones: {
+            create: {
+              status: 'NEW',
+            },
+          },
+        },
+      })
+
+      await expect(() =>
+        prisma.order.update({
+          where: {
+            orderId: orderId1,
+          },
+          data: {
+            orderId: orderId2,
+          },
+        }),
+      ).toThrowError()
```

Comment:
**This assertion is currently failing in CI.**

According to the solution here: https://www.notion.so/prismaio/emulated-onUpdate-Restrict-is-broken-e0e5109e23a64398896a748afdd8ddcd this should throw.

<img width="808" alt="Screenshot 2022-09-21 at 19 42 34" src="https://user-images.githubusercontent.com/35999252/191585135-83b75d49-83bc-4dfd-9039-ceffad5c51f3.png">

Reply:
@danstarns Note that this was incorrect 
```
      await expect(() =>
        prisma.order.update({
          where: {
            orderId: orderId1,
          },
          data: {
            orderId: orderId2,
          },
        }),
      ).toThrowError()
```
and this is the correct version
```
      await expect(
        prisma.order.update({
          where: {
            orderId: orderId1,
          },
          data: {
            orderId: orderId2,
          },
        }),
      ).rejects.toThrowError()
```